--- origsrc/tracker-0.6.4/src/libtracker/Makefile.am	2007-08-07 20:46:11.000000000 -0500
+++ src/tracker-0.6.4/src/libtracker/Makefile.am	2007-12-11 23:01:35.099238600 -0600
@@ -12,7 +12,7 @@
 	    	       $(DBUS_LIBS)
 
 libtrackerclient_la_SOURCES = tracker.c
-libtrackerclient_la_LDFLAGS = -version-info 0:0:0
+libtrackerclient_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
 tracker-client.h: $(top_srcdir)/data/tracker-introspect.xml Makefile.am
 	dbus-binding-tool --prefix=client_object --mode=glib-client --output=tracker-client.h $(top_srcdir)/data/tracker-introspect.xml
--- origsrc/tracker-0.6.4/src/libtracker-gtk/Makefile.am	2007-09-07 07:31:28.000000000 -0500
+++ src/tracker-0.6.4/src/libtracker-gtk/Makefile.am	2007-12-11 23:01:35.099238600 -0600
@@ -23,6 +23,6 @@
 	tracker-tag-bar.c	\
 	tracker-keyword-store.c
 
-libtracker_gtk_la_LIBADD = $(LIBTRACKERGTK_LIBS)
-libtracker_gtk_la_LDFLAGS = -version-info 0:0:0
+libtracker_gtk_la_LIBADD = ../libtracker/libtrackerclient.la $(LIBTRACKERGTK_LIBS)
+libtracker_gtk_la_LDFLAGS = -version-info 0:0:0 -no-undefined
 
--- origsrc/tracker-0.6.4/src/trackerd/trackerd.c	2007-12-10 20:58:25.000000000 -0600
+++ src/tracker-0.6.4/src/trackerd/trackerd.c	2007-12-11 23:01:35.114863600 -0600
@@ -37,6 +37,10 @@
 #include <glib/gi18n.h>
 #include <glib/gpattern.h>
 
+#ifdef __CYGWIN__
+#include <sys/file.h>
+#endif
+
 #include "config.h"
 
 #include "../xdgmime/xdgmime.h"
@@ -2618,7 +2622,11 @@
 		exit (1);
 	}
 
+#ifdef G_WITH_CYGWIN
+	if (flock (lfp, LOCK_EX) <0) {
+#else
 	if (lockf (lfp, F_TLOCK, 0) <0) {
+#endif
 		g_warning ("Tracker daemon is already running - attempting to run in readonly mode");
 		tracker->readonly = TRUE;
 	}
